src/resource.rs

```diff
@@ -323,6 +417,42 @@ pub struct TypeDef {
     pub tag: TypeDefTag,
 }
 
+/// Concrete instantiations of types and operations defined in resources.
+pub trait CustomConcrete: sealed::SealedConcrete {
```

Comment:
This is fine, but I wonder how much benefit there really is in having a trait to generalize over....things you might get by calling `to_custom` on one of the only-two-possible instances of `TypeParametrisedInternal`?

Reply:
What happens if you combine this with SealedConcrete i.e. hide the whole thing - can external folk still call its methods ?